### DISCUSSION TITLE
re-hydrate the cache from storage

### DISCUSSION THREAD

**Discussion by longstaff-elsevier** [TanStack/query#400] [General] [answered]
I am looking at a solution where we can hydrate the cache from a pre-known state, I can see how to extract the store through the subscribe method, but there is no way of populating the whole store again from that object that I can see.
Is this supported and/or planned as it should allow for isomorphic apps etc. or is this too complex with the hooks?

**Comment by tannerlinsley** [comment, marked as answer]
You can populate the cache using `queryCache.setQueryData(queryKey, data)`. With that primitive, you should be able to do most anything you need to, including looping over cache entries and using that to hydrate the cache.
